rocketpy/rocket/rocket.py

```diff
@@ -1119,6 +1148,90 @@ def add_parachute(
+            number and the third column is the drag coefficient. If a string,
+            it must be the path to a .csv or .txt file containing the drag
+            coefficient curve. The file must contain no headers and the first
+            column must specify the deployed level, the second column must
+            specify the Mach number and the third column must specify the drag
+            coefficient. If a Function, it must take as input the deployed
+            level and the Mach number and return the drag coefficient.
+        controller_function : function, callable
+            A function that takes the following arguments, in this order:
+
+            1. Time of the simulation at the current step in seconds.
+            2. The state vector of the simulation, which is defined as:
+
+               `[x, y, z, vx, vy, vz, e0, e1, e2, e3, wx, wy, wz]`.
+            3. A list containing the objects to be acted upon by the controller.
+               The objects in this list are the same as the objects in the
+               observed_objects list, but they can be modified by the controller.
+
+            This function will be called during the simulation at the specified
+            sampling rate. The function should evaluate and change the observed
+            objects as needed. The function should return None.
+
+            Note: The function will be called according to the sampling rate
+            specified.
+        sampling_rate : float
+            The sampling rate of the controller function in Hertz (Hz). This
+            means that the controller function will be called every
+            `1/sampling_rate` seconds.
+        reference_area : float, optional
+            Reference area used to calculate the drag force of the airbrakes
+            from the drag coefficient curve. If None, which is default, use
+            rocket section area. Must be given in squared meters.
+        name : string, optional
+            Airbrakes name, such as drogue and main. Has no impact in
+            simulation, as it is only used to display data in a more
+            organized matter.
+        controller_name : string, optional
+            Controller name. Has no impact in simulation, as it is only used to
+            display data in a more organized matter.
+
+        Returns
+        -------
+        airbrakes : Airbrakes
+            Airbrakes object created.
+        controller : Controller
+            Controller object created.
+        """
+        reference_area = reference_area if reference_area is not None else self.area
+        airbrakes = Airbrakes(
+            cd_curve=cd_curve,
+            reference_area=reference_area,
+            deployed_level=0,
+            name=name,
+        )
+        # save sampling rate on airbrakes so that it is available to inside
+        # controller function if needed
+        airbrakes.sampling_rate = sampling_rate
```

Comment:
This is a little weird, but I found no better way to make the user have access to the `sampling_rate` when defining its `controller_function`

Reply:
Maybe we could define the sampling rate as an argumnet of the `AirBrake` class (already using the new name sorry hahahaha).

One may suggest that the same aerosurface could be added to different rockets with different sampling rates...
But well, it turns out that the Parachutes also have their sampling rate as an argument!! So I think it would make more sense adding sampling_rate directly to the __init__ method of `AirBrake`

Reply:
It is a solution, but the thing is that `sampling_rate` is an argument for the `Controller`, which is a separate thing than the air brakes. It makes more sense for the controller to have the sampling rate than the air brakes

If we add `sampling_rate` to the `AirBrakes` init, then there will be two places where we have to pass the same `sampling_rate`, the `Controller` and the `AirBrakes`

The other way to get the `sampling_rate` inside the Controller function would be to have it as a input, meaning the controller function would have 4 inputs now (time, sampling_rate, state, list of objects).

I need opnions on this @Gui-FernandesBR @giovaniceotto @phmbressan

Reply:
@MateusStano did you solve this one already or not? Please ignore the next lines if you did.

I still don't understand the downsides of this option:

> The other way to get the sampling_rate inside the Controller function would be to have it as a input, meaning the controller function would have 4 inputs now (time, sampling_rate, state, list of objects).

Reply:
Yes, the sampling rate has been added as an argument to the controller function